runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeSearchPanel.java

```diff
@@ -26,46 +27,92 @@
 
 import com.google.common.base.Strings;
 import java.awt.BorderLayout;
-import java.awt.Color;
-import java.awt.GridLayout;
-import java.awt.image.BufferedImage;
+import java.awt.CardLayout;
+import java.awt.Dimension;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.ScheduledExecutorService;
 import javax.imageio.ImageIO;
-import javax.swing.Icon;
 import javax.swing.ImageIcon;
-import javax.swing.JLabel;
 import javax.swing.JPanel;
+import javax.swing.JScrollPane;
 import javax.swing.SwingUtilities;
 import javax.swing.border.EmptyBorder;
 import lombok.extern.slf4j.Slf4j;
 import net.runelite.api.Client;
 import net.runelite.api.ItemComposition;
 import net.runelite.client.game.AsyncBufferedImage;
 import net.runelite.client.game.ItemManager;
-import net.runelite.client.plugins.hiscore.IconTextField;
+import net.runelite.client.ui.ColorScheme;
+import net.runelite.client.ui.components.IconTextField;
+import net.runelite.client.ui.components.PluginErrorPanel;
 import net.runelite.http.api.item.Item;
+import net.runelite.http.api.item.ItemClient;
 import net.runelite.http.api.item.ItemPrice;
 import net.runelite.http.api.item.SearchResult;
 
+/**
+ * This panel holds the search section of the Grand Exchange Plugin.
+ * It should display a search bar and either item results or a label (Searching..)
+ */
 @Slf4j
 class GrandExchangeSearchPanel extends JPanel
 {
-	private static final List<GrandExchangeItems> ITEMS_LIST = new ArrayList<>();
-
 	private final Client client;
 	private final ItemManager itemManager;
 	private final ScheduledExecutorService executor;
 
-	private Icon search;
+	private final ItemClient itemClient = new ItemClient();
```

Comment:
what is this for?

Reply:
ItemClient? I'm not sure, I wasn't the one who implemented the fetching.